webapp/src/ts/effects/reports.effects.ts

```diff
@@ -128,12 +128,20 @@ export class ReportsEffects {
     return this.actions$.pipe(
       ofType(ReportActionList.markReportRead),
       concatMap(action => of(action).pipe(
-        withLatestFrom(this.store.select(Selectors.getListReport, { id: action?.payload?.id })),
+        withLatestFrom(
+          this.store.select(Selectors.getListReport, { id: action?.payload?.id }),
+          this.store.pipe(select(Selectors.getUnreadCount))
+        ),
       )),
-      exhaustMap(([action, report]) => {
+      exhaustMap(([action, report, unreadCount]) => {
         if ((report && report.read) || !action?.payload?.id) {
           return of();
         }
+
+        if (unreadCount) {
+          this.globalActions.updateUnreadCount({ report: unreadCount.report - 1 });
```

Comment:
I'm wondering if it would be nicer if the action would be like `updateUnreadCount({ report: -1 })` and then the reducer would just update the value with the diff. Then we wouldn't need to get the value here in the 1st place. 

It's fine to keep it like this (like the old code).

Reply:
Oh yeah, I like better that way. Maybe next time, for now I'll keep it as the old way. :)